youtube_dl/extractor/dplay.py

```diff
@@ -5,40 +5,70 @@
 
 from .common import InfoExtractor
 from ..utils import int_or_none
-
+from ..compat import compat_urlparse
 
 class DPlayIE(InfoExtractor):
-    _VALID_URL = r'http://www\.dplay\.se/[^/]+/(?P<id>[^/?#]+)'
-
-    _TEST = {
-        'url': 'http://www.dplay.se/nugammalt-77-handelser-som-format-sverige/season-1-svensken-lar-sig-njuta-av-livet/',
```

Comment:
Didn't notice you placed it second. Anyway it fails after this changes:
```
> py26yt -v test:DPlay_1
[debug] System config: []
[debug] User config: []
[debug] Command-line args: [u'-v', u'test:DPlay_1']
[debug] Encodings: locale cp1251, fs mbcs, out cp866, pref cp1251
[debug] youtube-dl version 2016.02.22
[debug] Git HEAD: 73ab2dd
[debug] Python version 2.6.6 - Windows-2003Server-5.2.3790-SP2
[debug] exe versions: avconv 11.3, avprobe 11.3, ffmpeg N-77883-gd7c75a5, ffprobe N-77883-gd7c75a5, rtmpdump 2.4
[debug] Proxy map: {}
[TestURL] Test URL: http://www.dplay.se/nugammalt-77-handelser-som-format-sverige/season-1-svensken-lar-sig-njuta-av-livet/
[DPlay] season-1-svensken-lar-sig-njuta-av-livet: Downloading webpage
[DPlay] 3172: Downloading JSON metadata
[DPlay] 3172: Downloading m3u8 information
ERROR: Failed to download m3u8 information: HTTP Error 403: Forbidden (caused by HTTPError()); please report this issue on https://yt-dl.org/bug . Make sure you
 are using the latest version; see  https://yt-dl.org/update  on how to update. Be sure to call youtube-dl with the --verbose flag and include its complete outp
ut.
  File "C:\Dev\git\youtube-dl\master\youtube_dl\extractor\common.py", line 355, in _request_webpage
    return self._downloader.urlopen(url_or_request)
  File "C:\Dev\git\youtube-dl\master\youtube_dl\YoutubeDL.py", line 1905, in urlopen
    return self._opener.open(req, timeout=self._socket_timeout)
  File "C:\python\python266\lib\urllib2.py", line 397, in open
    response = meth(req, response)
  File "C:\python\python266\lib\urllib2.py", line 510, in http_response
    'http', request, response, code, msg, hdrs)
  File "C:\python\python266\lib\urllib2.py", line 435, in error
    return self._call_chain(*args)
  File "C:\python\python266\lib\urllib2.py", line 369, in _call_chain
    result = func(*args)
  File "C:\python\python266\lib\urllib2.py", line 518, in http_error_default
    raise HTTPError(req.get_full_url(), code, msg, hdrs, fp)
```
Before this change:
```
> py26yt -v test:DPlay
[debug] System config: []
[debug] User config: []
[debug] Command-line args: [u'-v', u'test:DPlay']
[debug] Encodings: locale cp1251, fs mbcs, out cp866, pref cp1251
[debug] youtube-dl version 2016.02.22
[debug] Git HEAD: 73ab2dd
[debug] Python version 2.6.6 - Windows-2003Server-5.2.3790-SP2
[debug] exe versions: avconv 11.3, avprobe 11.3, ffmpeg N-77883-gd7c75a5, ffprobe N-77883-gd7c75a5, rtmpdump 2.4
[debug] Proxy map: {}
[TestURL] Test URL: http://www.dplay.se/nugammalt-77-handelser-som-format-sverige/season-1-svensken-lar-sig-njuta-av-livet/
[DPlay] season-1-svensken-lar-sig-njuta-av-livet: Downloading webpage
[DPlay] 3172: Downloading JSON metadata
[DPlay] 3172: Getting manifest url for hls stream
[DPlay] 3172: Downloading m3u8 information
[debug] Invoking downloader on u'http://sbsvod02-vh.akamaihd.net/i/delivery/9b/64/9b647144-0347-4a96-9c91-255f82d7bde9/SE304349146527_,25,2,6,10,15,00k.mp4.csmi
l/index_0_av.m3u8?null='
[hlsnative] Downloading m3u8 manifest
[hlsnative] Total fragments: 265
[download] Destination: Svensken lr sig njuta av livet-3172.mp4
[download]   1.9% of ~888.61MiB at Unknown speed ETA Unknown ETA
ERROR: Interrupted by user
```